spark/src/test/java/org/apache/iceberg/spark/source/TestParquetWrite.java

```diff
@@ -163,7 +164,8 @@ public void testAppend() throws IOException {
     Assert.assertEquals("Result rows should match", expected, actual);
   }
 
-  @Test
+  // ignore due to spark default use static
+  @Ignore
```

Comment:
Shouldn't this change the Spark configuration to dynamic mode?

Reply:
After looking into this more, the problem is that the `DataFrameWriter` API imposes truncate behavior for all DSv2 implementations using `SaveMode.Overwrite`. This matches the built-in file sources and the idea is to standardize behavior on those.

I'm not sure if dynamic mode would make a difference in v1, so we should check that and add it here if it is respected: https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala#L274-L279

For now, I agree with @chenjunjiedada's approach and added a write option to override the mode. If you set "overwrite-mode" to "dynamic" then it will use the Spark 2.4 behavior.